core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -545,6 +559,50 @@ default void finishUpdate(ConnectorSession session, ConnectorTableHandle tableHa
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support updates");
     }
 
+    /**
+     * Return the row update paradigm supported by the connector on the table, or
+     * {@link RowChangeParadigm#ROW_CHANGE_NOT_SUPPORTED} if row change is not supported.
+     * @param session The session in which to start the update operation.
+     * @param tableHandle A ConnectorTableHandle for the table to be updated.
+     * @return Return the supported row update paradigm.
+     */
+    default RowChangeParadigm getRowChangeParadigm(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return ROW_CHANGE_NOT_SUPPORTED;
+    }
+
+    /**
+     * Get the column handle that will generate row IDs for the merge operation.
+     * These IDs will be passed to the {@code storeMergedRows()} method of the
+     * {@link io.trino.spi.connector.ConnectorMergeSink} that created them.
```

Comment:
You can link to a method `{@link ConnectorMergeSink#storeMergedRows()}`

same below

Reply:
Done.